pandas/core/indexes/datetimes.py

```diff
@@ -56,12 +53,48 @@ def _new_DatetimeIndex(cls, d):
             warnings.simplefilter("ignore")
             result = cls.__new__(cls, verify_integrity=False, **d)
 
-    if tz is not None:
-        result = result.tz_localize('UTC').tz_convert(tz)
     return result
 
 
-class DatetimeIndex(DatetimeArray, DatetimeIndexOpsMixin, Int64Index):
+class DatetimeDelegateMixin(DatetimelikeDelegateMixin):
+    _extra_methods = [
+        'to_perioddelta',
+        'to_julian_date',
+    ]
+    _extra_raw_methods = [
```

Comment:
Is it explained somewhere what the 'raw' methods mean (and in general all the different sets here)?

Reply:
This is all documented in DatetimelikeDelegateMixin.

```
    * raw_methods : Set
        The set of methods whose results should should *not* be
        boxed in an index, after being returned from the array
    * raw_properties : Set
        The set of properties whose results should should *not* be
        boxed in an index, after being returned from the array
```

I don't recall if I explained this, but I briefly added these `extra_*` item to the `datelike_ops` on the array classes. I ended up reverting that change though, because they were automatically picked up by the `Series.dt` accessor, and I don't think we want some of these like `to_peridoddelta` in that API.

Reply:
Let me know if there's anything I could do to clarify this.

Personally, I find all this indirection confusing, but worry that additional comments in each of the subclasses would just be noise.

Reply:
can you basically repeat the above in a comment here. I agree this is not obvious.